### DISCUSSION TITLE
kzg changes

### DISCUSSION THREAD

**Discussion by HIT2022** [wevm/viem#2809] [Idea / Feature Request]
### Check existing issues

- [x] I checked there isn't [already an issue](https://github.com/wevm/viem/issues) for the bug I encountered.

### Viem Version

2.21.16

### Current Behavior

`kzg-wasm` released 0.5.0 with the major changeset coming from this PR: https://github.com/ethereumjs/kzg-wasm/pull/17. You can see how they renamed `blobToKzgCommitment` to `blobToKZGCommitment` and `computeBlobKzgProof` to `computeBlobKzgProof`. Reporting as a bug so that `blobsToProofs` and `blobsToCommitments` can be updated to account for both versions of these APIs.

On this subject: [micro-eth-signer](https://github.com/paulmillr/micro-eth-signer/) added kzg support so there is now a native JS implementation of KZG cryptography. It seems generation of commitments, in particular, is significantly faster than `kzg-wasm`. Could we refactor kzg use so that viem also supports this library's functions (`blobToKzgCommitment` and `computeBlobKZGProof`)?

Unrelated question: could `sendTransaction` handling potentially be adjusted to optionally support passing of _pre-computed_ sidecar data? For UX purposes, if/when multiple blobs are in play, I'd like to be able to execute the cryto well in advance of submitting the tx. to the network.

### Expected Behavior

Viem's use of kzg will work with `kzg-wasm` release 0.5.0

### Steps To Reproduce

Use viem with `kzg-wasm` release 0.5.0

**Comment by jxom** [comment]
> You can see how they renamed blobToKzgCommitment to blobToKZGCommitment and computeBlobKzgProof to computeBlobKzgProof. Reporting as a bug so that blobsToProofs and blobsToCommitments can be updated to account for both versions of these APIs.

Will move as a discussion as a feature request to add support for hex i/o intefaces. For now, you could use `defineKzg` to convert hex -> uint8array.

> could sendTransaction handling potentially be adjusted to optionally support passing of pre-computed sidecar data?

You can use the `sidecars` property on `sendTransaction` (need to document this).
